<details> <summary><H2> Quiime2 Projects </H2></summary>

<details> <summary><H3> Human microbiome Study </H3></summary>
  
Data taken from [this](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-016-0225-7) Human microbiome study is used to perform a bioinformatic pathway analysis based on [this](https://docs.qiime2.org/2022.2/tutorials/fmt/) tutorial by qiime2. 

Some stuff you will get done:

- cleaning + assessment of raw read inputs
- alignment of 16s regions
- classification of microbes
- phylogenetic tree visualization of microbe relationships
- Diversity metrics of dataset

<details> <summary><i> initial data downloads </i></summary>

[sample metadata](https://data.qiime2.org/2022.2/tutorials/fmt/sample_metadata.tsv)  

<br>

[foward reads](https://data.qiime2.org/2022.2/tutorials/fmt/fmt-tutorial-demux-1-10p.qza)  

<br>

[reverse reads](https://data.qiime2.org/2022.2/tutorials/fmt/fmt-tutorial-demux-2-10p.qza)
  
</details> <!-- end initial data downloads -->

<details> <summary><i> an example output </i></summary>

![](https://github.com/Kale-23/Qiime2_Microbiome_Analysis/blob/main/plots/alpha-group-sig-obs-feats.png)

Shows an alpha diversity metric on the y axis, and a metadata variable on the x. This specific plot shows the observed features metric vs treatment group.

</details>  <!-- End an example output -->

</details> <!-- End Human microbiome Study -->

</details> <!-- End Quiime2 Projects -->
